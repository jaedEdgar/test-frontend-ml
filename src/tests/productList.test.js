import React from 'react';
import renderer from 'react-test-renderer';
import ProductList from "../client/components/productList"
import '@testing-library/jest-dom'
import { MemoryRouter as Router, withRouter } from 'react-router-dom'

const mock = [
  {
    city_name: "Colegiales",
    condition: "new",
    free_shipping: true,
    id: "MLA908778425",
    picture: "http://http2.mlstatic.com/D_771888-MLA43752069008_102020-I.jpg",
    price: {
      amount: "224990",
      currency: "ARS",
      decimals: 0
    },
    title: "Asus Rog Phone 3 Zs661ks Dual Sim 256 Gb Negro Brillante 12 Gb Ram"
  }, {
    city_name: "Colegiales",
    condition: "new",
    free_shipping: true,
    id: "MLA908778425",
    picture: "http://http2.mlstatic.com/D_771888-MLA43752069008_102020-I.jpg",
    price: {
      amount: "224990",
      currency: "ARS",
      decimals: 0
    },
    title: "Asus Rog Phone 3 Zs661ks Dual Sim 256 Gb Negro Brillante 12 Gb Ram"
  }
]




it('will pass', () => {
  expect(
    renderer.create(<Router>
      <ProductList products={mock} />
    </Router>)
      .toJSON()
  ).toMatchSnapshot()
})