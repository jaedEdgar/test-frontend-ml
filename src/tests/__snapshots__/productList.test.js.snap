// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will pass 1`] = `
<div
  className="container"
>
  <ul
    className="result-search"
  >
    <li
      className="result-search--item"
    >
      <div
        className="result-search--preview"
      >
        <img
          alt="Asus Rog Phone 3 Zs661ks Dual Sim 256 Gb Negro Brillante 12 Gb Ram"
          src="http://http2.mlstatic.com/D_771888-MLA43752069008_102020-I.jpg"
        />
      </div>
      <div
        className="result-search--wrapp"
      >
        <div
          className="result-search--description"
        >
          <div
            className="result-search--description-price"
          >
            <span
              className="result-search--description-mount"
            >
              $ 
              224,990
            </span>
            <i
              className="free-shipping-icon"
              title="Envío gratis"
            />
          </div>
          <h2>
            <a
              className="result-search--description-title"
              href="/items/MLA908778425"
              onClick={[Function]}
            >
              Asus Rog Phone 3 Zs661ks Dual Sim 256 Gb Negro Brillante 12 Gb Ram
            </a>
          </h2>
        </div>
        <span
          className="result-search--description-city"
        >
          Colegiales
        </span>
      </div>
    </li>
    <li
      className="result-search--item"
    >
      <div
        className="result-search--preview"
      >
        <img
          alt="Asus Rog Phone 3 Zs661ks Dual Sim 256 Gb Negro Brillante 12 Gb Ram"
          src="http://http2.mlstatic.com/D_771888-MLA43752069008_102020-I.jpg"
        />
      </div>
      <div
        className="result-search--wrapp"
      >
        <div
          className="result-search--description"
        >
          <div
            className="result-search--description-price"
          >
            <span
              className="result-search--description-mount"
            >
              $ 
              224,990
            </span>
            <i
              className="free-shipping-icon"
              title="Envío gratis"
            />
          </div>
          <h2>
            <a
              className="result-search--description-title"
              href="/items/MLA908778425"
              onClick={[Function]}
            >
              Asus Rog Phone 3 Zs661ks Dual Sim 256 Gb Negro Brillante 12 Gb Ram
            </a>
          </h2>
        </div>
        <span
          className="result-search--description-city"
        >
          Colegiales
        </span>
      </div>
    </li>
  </ul>
</div>
`;
